### PR TITLE
[5940] Consider repository path in components datasource when creating components

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5940

### CODE REVIEW

File: ui/app/src/utils/content.ts [hunk 1]
```diff
@@ -1009,3 +1009,10 @@ export const openItemEditor = (
     );
   }
 };
+
+export function generateComponentBasePath(contentType: string) {
+  return `/site/components/${contentType.replace('/component/', '')}s/`.replace(/\/{1,}$/m, '');
+}
```
Comment (rart) [inline ui/app/src/utils/content.ts:1015]:
```suggestion
}

```

File: ui/legacy/src/components/cstudio-forms/data-sources/components.js [hunk 2]
```diff
@@ -30,7 +30,8 @@
     this.allowEmbedded = false;
     this.enableSearch = false;
     this.enableBrowse = false;
-    this.baseRepoPath = '/site/components';
+    this.defaultBaseRepoPath = '/site/components';
+    this.baseRepoPath = null;
     this.baseBrowsePath = '/site/components';
```
Comment (rart) [inline ui/legacy/src/components/cstudio-forms/data-sources/components.js:35]: Does the base browse path work correctly?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/legacy/src/components/cstudio-forms/data-sources/components.js (hunk 2 above)
Comment (jvega190) [reply ui/legacy/src/components/cstudio-forms/data-sources/components.js:35]: yes it does, it gets overridden later in the code (if set in content-type editor prop)

**Comment by rart** [comment]
Test form with more than one components data source on the same item selector making sure the embedded and shared rules work correctly (review ticket desc), paths for browse/search and repo path all work correctly

**Review by rart — APPROVED** [review]

**Pull request merged**